python/ray/train/v2/torch/torch_trainer.py

```diff
@@ -194,9 +197,22 @@ def __init__(
         # TODO: [Deprecated]
         metadata: Optional[Dict[str, Any]] = None,
         resume_from_checkpoint: Optional[Checkpoint] = None,
+        running_without_ray_train_controller: Optional[bool] = False,
```

Comment:
I think this will be a bit confusing for users because the Train Controller is not a public concept that's exposed to the user.

I've been thinking more about this and think it should be tied to the `ScalingConfig`, have we discussed the option of executing this code path with `ScalingConfig.num_workers=0`?